### DISCUSSION TITLE
Add an option to skip table creation in flex output

### DISCUSSION THREAD

**Discussion by nextstopsun** [osm2pgsql-dev/osm2pgsql#1493] [General]
Database tables may have generated columns, triggers, rules, constraints etc., that currently can't be created in lua scripts.
An option to use existing table instead of recreating it opens up a lot of SQL possibilities. 
This option doesn't seem like a lot of additional code.

**Comment by joto** [comment]
osm2pgsql interacts with the database in many ways, so it has to know about the tables, what columns they have etc. so it can fill them correctly, build the right indexes, etc. In theory it might be nice to have an option to say: Please osm2pgsql get out of my hair, just fill those tables, I'll do the rest. But in practice this means you would need to know a lot about how osm2pgsql interacts with the database, set up everything just so it is exactly right. And there are many subtle ways how this could fail and osm2pgsql couldn't even tell you what the problem is. There are only very few users who are actually at that level, so this isn't something that would actually help that many users and it would probably generate a lot of questions and support effort on the part of the maintainers.

**Reply by nextstopsun** [reply]
I propose to leave all logic as is. Let flex know the table structure from lua code. Just add a little flag to trust that everything is in place already.
DB will throw an error if anything is wrong and that's ok imho.
That's not an osm2pgsql option, it's just a flag in flex config lua script.

**Comment by nextstopsun** [comment]
As an important addition, table partitioning can't be set up before first import either. 
And if your data is big enough it's quite painful to partition after it's already written to the DB in one piece.

In reality osm2pgsql already has a similar flag (which is `append` mode). If you break tables in the db after initial import, append will fail - and that's an expected behaviour. Same goes to appending to a clear (or truncated) table. You don't really need initial import in `create` mode if everything (all used functions and tables) is created as needed in the db.

**Comment by lonvia** [comment]
Duplicate of #763 and #758. The answer still applies.
